src/ext/java/org/opentripplanner/ext/legacygraphqlapi/datafetchers/LegacyGraphQLAlertImpl.java

```diff
@@ -126,22 +127,22 @@ public DataFetcher<Iterable<Map.Entry<String, String>>> alertUrlTranslations() {
   @Override
   public DataFetcher<Long> effectiveEndDate() {
     return environment -> {
-      Date effectiveEndDate = getSource(environment).getEffectiveEndDate();
+      Instant effectiveEndDate = getSource(environment).getEffectiveEndDate();
       if (effectiveEndDate == null) {
         return null;
       }
-      return effectiveEndDate.getTime() / 1000;
+      return effectiveEndDate.getEpochSecond();
```

Comment:
Other parts of the legacy API use epochMillis: https://github.com/opentripplanner/OpenTripPlanner/blob/dev-2.x/src/ext/java/org/opentripplanner/ext/legacygraphqlapi/datafetchers/LegacyGraphQLLegImpl.java#L66

Which one is correct? I noticed this _after_ I added the vehicle position endpoint where I also used epochSeconds.

Reply:
This preserves the old behaviour, which seems to be inconsistent. In the new GraphQL API, we want to expose everything as ISO-8601 strings